src/dsp_tools/commands/start_stack.py

```diff
@@ -112,6 +118,41 @@ def _copy_resources_to_home_dir(self) -> None:
             shutil.copy(file_path, self.__docker_path_of_user / file.name)
         if not self.__stack_configuration.latest_dev_version:
             Path(self.__docker_path_of_user / "docker-compose.override.yml").unlink()
+        if self.__stack_configuration.custom_host is not None:
+            Path(self.__docker_path_of_user / "dsp-app-config.json").unlink()
+
+    def _set_custom_host(self) -> None:
+        """
+        To ensure the frontend can communicate with a backend on a different server, the host in the environments
+        needs to be changed.
+        By design the IRIs match the host of the database.
+
+        This is done by overriding the environment variables in the docker-compose.yml and by replacing the
+        configuration for the frontend.
+        """
+
+        if self.__stack_configuration.custom_host is not None:
+            self.__localhost_url = f"http://{self.__stack_configuration.custom_host}"
+
+            docker_template_path = importlib.resources.files("dsp_tools").joinpath(
+                "resources/start-stack/docker-compose.override-host.j2"
+            )
+            docker_template = Template(docker_template_path.read_text(encoding="utf-8"))
+            docker_template_rendered = docker_template.render(CUSTOM_HOST=self.__stack_configuration.custom_host)
+            Path(self.__docker_path_of_user / "docker-compose.override-host.yml").write_text(
+                docker_template_rendered, encoding="utf-8"
+            )
+
+            dsp_app_config_template_path = importlib.resources.files("dsp_tools").joinpath(
+                "resources/start-stack/dsp-app-config.override-host.j2"
+            )
+            dsp_app_config_template = Template(dsp_app_config_template_path.read_text(encoding="utf-8"))
+            dsp_app_config_rendered = dsp_app_config_template.render(CUSTOM_HOST=self.__stack_configuration.custom_host)
+            Path(self.__docker_path_of_user / "dsp-app-config.json").write_text(
+                dsp_app_config_rendered, encoding="utf-8"
+            )
```

Comment:
To make it clearer what happens, I would delete the existing config file directly before writing to it. Currently, you delete it on line 122. That works, but then we have 2 things at 2 different places, while they belong together.

Reply:
Good point. Moving it even saves one `if` statement.